packages/environment/test/connectOrStartTest.js

```diff
@@ -0,0 +1,51 @@
+const assert = require("chai").assert;
+const Develop = require("../develop");
+
+describe("connectOrStart test network", async function () {
+  const ipcOptions = { network: "test" };
+  const ganacheOptions = {
+    host: "127.0.0.1",
+    network_id: 666,
+    port: 6969
+  };
+
+  it("starts Ganache when no Ganache instance is running", async function () {
+    let connection;
+    try {
+      connection = await Develop.connectOrStart(ipcOptions, ganacheOptions);
+      assert.isTrue(connection.started, "A new Ganache server did not spin up");
+      assert.isFunction(connection.disconnect, "disconnect is not a function");
+    } finally {
+      if (connection) {
+        connection.disconnect();
+      }
+    }
+  });
+
+  it("connects to an established Ganache instance", async function () {
+    let connectionOneDisconnect, connectionTwoDisconnect;
+
+    try {
+      //establish a connection
```

Comment:
Does this starts the connection and disconnect it?

Reply:
I'm not sure what you're asking. This test verifies subsequent calls to `Develop.connectOrStart` after a connection has been established, will connect instead of starting the underlying service. It does start a connection and eventually cleans it up in the `finally` block at the end of the test.

Reply:
ah ok. got it. thank you.